File: Python/345__Reverse_Vowels_of_a_String.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = ("a", "e", "i", "o", "u", "A", "E", "I", "O", "U")
        i, j = 0, len(s) - 1
        lst = [c for c in s]

        while i < j:
            while  i < j  and lst[i] not in vowels:
                i += 1
            while  i < j and lst[j] not in vowels:
                j -= 1
            lst[i], lst[j] = lst[j], lst[i]
            i += 1
            j -= 1

        return "".join(lst)

File: Java/345__Reverse_Vowels_of_a_String.java
class Solution {
    public String reverseVowels(String s) {
        int i = 0, j = s.length() - 1;
        Set<Character> vowel = new HashSet<>();
        vowel.add('a');
        vowel.add('e');
        vowel.add('i');
        vowel.add('o');
        vowel.add('u');
        vowel.add('A');
        vowel.add('E');
        vowel.add('I');
        vowel.add('O');
        vowel.add('U');
        char[] s_arr = s.toCharArray();
        while (i < j) {
            while (i < j && !vowel.contains(s_arr[i])) i++;
            while (i < j && !vowel.contains(s_arr[j])) j--;
            char temp = s_arr[i];
            s_arr[i] = s_arr[j];
            s_arr[j] = temp;
            i++;
            j--;
        }
        return String.valueOf(s_arr);

    }
}
